comb_spec_searcher/strategies/rule.py

```diff
@@ -543,18 +546,25 @@ def constructor(self) -> DisjointUnion:
                 k: k for k in self.comb_class.extra_parameters
             }
             for rule in self.rules:
+                # TODO: call extra parameters on Rule class
                 if isinstance(rule, EquivalenceRule):
                     rules_parameters = rule.strategy.extra_parameters(
                         rule.comb_class, rule.actual_children
                     )[rule.child_idx]
+                elif isinstance(rule, ReverseRule):
+                    forward_rule_parameters = rule.strategy.extra_parameters(
+                        rule.children[0]
+                    )[0]
+                    rules_parameters = {
+                        b: a for a, b in forward_rule_parameters.items()
+                    }
                 else:
                     rules_parameters = rule.strategy.extra_parameters(
                         rule.comb_class, rule.children
                     )[0]
                 extra_parameters = {
-                    rules_parameters[child_var]: parent_var
-                    for child_var, parent_var in extra_parameters.items()
-                    if child_var in rules_parameters
+                    rules_parameters[parent_var]: child_var
+                    for parent_var, child_var in extra_parameters.items()
```

Comment:
This really gets flipped for both EquivalenceRule and ReverseRule?

Reply:
This is when we have some path 
A = B = C 
but the actual strategy we have is A = B and C = B, so I think it is correct.
parent in A maps to child in B, then the parent in B (which is a child in C = B) maps to the child on C (which is parent on the rule B = C). Or something.